metals/src/main/scala/scala/meta/internal/metals/ServerLivenessMonitor.scala

```diff
@@ -178,17 +134,25 @@ object ServerLivenessMonitor {
   object FirstPing extends State
   object Running extends State
 
-  class ReconnectOptions(optReconnect: Option[() => Future[Unit]]) {
-    val canReconnect = optReconnect.nonEmpty
-    val reconnect: () => Future[Unit] =
-      optReconnect.getOrElse(() => Future.successful(()))
-  }
-
-  object ReconnectOptions {
-    def empty = new ReconnectOptions(None)
-    def apply(reconnect: () => Future[Unit]) = new ReconnectOptions(
-      Some(reconnect)
-    )
-  }
+  def connectedParams(serverName: String, icons: Icons): MetalsStatusParams =
+    MetalsStatusParams(
+      s"$serverName ${icons.link}",
+      "info",
+      show = true,
+      tooltip = s"Metals is connected to the build server ($serverName).",
+    ).withStatusType(StatusType.bsp)
+
+  val disconnectedParams: MetalsStatusParams =
+    MetalsStatusParams("", hide = true).withStatusType(StatusType.bsp)
+
+  def noResponseParams(serverName: String, icons: Icons): MetalsStatusParams =
+    MetalsStatusParams(
+      s"$serverName ${icons.error}",
+      "error",
+      show = true,
+      tooltip =
+        s"Broken connection, build sever ($serverName) is not responding. Press to reconnect.",
```

Comment:
Question about the "Press to reconnect" part of the tooltip: it will work great for the VSCode, but about the other LSP clients, such as the Neovim with the `nvim-metals` plugin? It could expose the BSP status bar so that it would be shown in the user's statusline, but the "press" action probably won't be available. Maybe we could make the message more generic, WDYT? :slightly_smiling_face:

Reply:
Hmm... right.

Reply:
I added additional field  `commandTooltip`.

Reply:
Thank you! :+1: